plugin/CalligraDocumentPage.qml

```diff
@@ -50,14 +50,25 @@ DocumentPage {
     onStatusChanged: {
         //Delay loading the document until the page has been activated.
         if (status === PageStatus.Active) {
-            document.source = page.source
+            asyncLoading.restart()
         }
     }
 
+
+    backNavigation: !busy
     icon: "image://theme/icon-m-file-formatted"
     busy: doc.status !== Calligra.DocumentStatus.Loaded
             && doc.status !== Calligra.DocumentStatus.Failed
 
+    Timer {
+        id: asyncLoading
+        interval: 1
+        onTriggered: {
```

Comment:
Either use Qt.callLater() in onStatusChanged so the functionality is localised there,  or use a running: doc.status === Calligra.DocumentStatus.Loaded binding so it all here.

Reply:
> Qt.callLater()

AFAIK Not yet available in Qt 5.6

> use a running: doc.status === Calligra.DocumentStatus.Loaded 

Done.